### ISSUE TITLE
How to cancel decompression?

### ISSUE THREAD

**Issue by jardik** [conor42/fast-lzma2#8]
Is there correct way to cancel decompression of the stream? I am using similar code as in the file_test.c file. i.e. I call FL2_createDStreamMt once at start and then for each stream stored in a file I call FL2_initDStream and then keep calling FL2_decompressStream until I decompress all data. But if my file reading function returns an error, what should I call to cancel the decompression so that next FL2_initDStream doesn't return an error (unfinished decompression of previous stream)? It seems the example in file_test.c doesn't handle reading errors and they are just ignored. I found FL2_cancelDStream, is this the function I should call?

**Comment by conor42** [comment]
Yes, FL2_cancelDStream() should be called. It's only required for multi-threaded decompression, but should probably be called for single-threaded too in case something changes. At the moment for ST you can init a new stream without canceling. I'll leave this open to remind me to add read error handling in file_test.

**Comment by jardik** [comment]
Thank you
